swiftsimio/subset_writer.py

```diff
@@ -0,0 +1,172 @@
+"""
+Contains functions for reading a subset of a SWIFT dataset and writing
+it to a new file.
+"""
+
+from swiftsimio.reader import SWIFTUnits, SWIFTMetadata
+from swiftsimio.masks import SWIFTMask
+from swiftsimio.accelerated import read_ranges_from_file
+
+import unyt
+import h5py
+import numpy as np
+
+def compute_mask_size(mask: SWIFTMask, dataset_name: str):
+    """
+    Return number of elements of the specified dataset selected by the mask
+
+    Parameters
+    ----------
+    mask : SWIFTMask
+        the mask used to define subset that is written to new snapshot
+    dataset_name : str
+        the name of the dataset we're interested in. This is the name from the
+        hdf5 file (i.e. "PartType0", rather than "gas")
+
+    Returns
+    -------
+    int
+        size of the subset of the dataset selected by the mask
+    """
+    legend = {"Cells": 0, 
+              "PartType0": mask.gas_size if hasattr(mask, "gas_size") else 0, 
```

Comment:
You should be able to use `swiftsimio.metadata.particle.particle_name_underscores` here.